Tests/EventListener/ReportSubscriberTest.php

```diff
@@ -0,0 +1,315 @@
+<?php
+
+namespace EventListener;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Mautic\ChannelBundle\Helper\ChannelListHelper;
+use Mautic\LeadBundle\Model\CompanyReportData;
+use Mautic\LeadBundle\Segment\Query\Expression\ExpressionBuilder;
+use Mautic\ReportBundle\Event\ReportBuilderEvent;
+use Mautic\ReportBundle\Helper\ReportHelper;
+use MauticPlugin\LeuchtfeuerCompanySegmentsBundle\Entity\CompanySegment;
+use MauticPlugin\LeuchtfeuerCompanySegmentsBundle\Entity\CompanySegmentRepository;
+use MauticPlugin\LeuchtfeuerCompanySegmentsBundle\EventListener\ReportSubscriber;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class ReportSubscriberTest extends TestCase
+{
+    private ReportSubscriber $reportSubscriber;
+    /** @var MockObject|ReportBuilderEvent */
+    private $reportBuilderEventMock;
+    private MockObject $companyReportDataMock;
```

Comment:
Does it really make sense to use union type for $companyReportDataMock. I just used it for $reportBilderEventMock because it's a partial mock but $companyReportDataMock will always be 100% a mockobject

Reply:
It's still a mock of the particular object. Will help now with the autocomplete, will help later when 8.3+ will be required.